kbase-extension/static/kbase/js/common/lang.js

```diff
@@ -23,8 +23,53 @@ define(['bluebird'], (Promise) => {
         });
     }
 
+    /**
+     * Converts a string to an integer.
+     * If it's not a string (or an integer already), this throws an Error.
+     * Floats are not reduced into integers, unless the decimal part === 0.
+     * @param {string|number} value
+     * @returns {number} the integer version of the value.
+     * @throws {*} an error if:
+     * - the value is a non-integer number,
+     * - if the value is a non-int-parseable string,
+     * - if value is anything else (like an Array or Object)
+     */
+    function toInteger(value) {
+        switch (typeof value) {
+            case 'number':
+                if (value !== Math.floor(value)) {
+                    throw new Error('Integer is a non-integer number');
+                }
+                return value;
+            case 'string':
+                if (value.match(/^[-+]?[\d]+$/)) {
+                    return parseInt(value, 10);
+                }
+                throw new Error('Invalid integer format');
+            default:
+                throw new Error('Type ' + typeof value + ' cannot be converted to integer');
+        }
+    }
+
+    /**
+     * Returns true if the value is an empty string (or entirely whitespace), or null.
+     * Returns false otherwise.
+     * @param {*} value
+     */
+    function isEmptyString(value) {
```

Comment:
We have a `string` module in `util/string` -- not sure if this should have gone in there? I don't know what the logic (if any) behind `string` being under `util` and this module being in `common` is.

Reply:
No idea. This is a generic util module, too, so it should probably go in `util`, and that would make more sense in `util/string`. How about I move `isEmptyString` to `util/string` now, and in another PR move `common/lang` -> `util/lang` (or maybe another name that makes more sense?). I'd like to avoid touching every file that uses this module in this PR.

Reply:
Ok, done. moved `isEmptyString` to `util/string`.